automated-testing/ui-testing/readme.md

```diff
@@ -0,0 +1,64 @@
+# UI Testing
+
+This section is primarily geared towards web-based UIs, but the guidance is similar for mobile and OS based applications.  
+
+## Goals
+
+UI testing provides the ability to ensure that users have a consistent visual user experience across a variety of means of access and that the user interaction is consistent with the function requirements.
+
+### UI tests have several goals
+
+- Ensure the UI appearance and interaction satisfy the functional and non-functional requirements
+- Detect changes in the UI both across devices and delivery platforms and between code changes
+- Provide confidence to designers and developers the user experience is consistent
+- Support fast code evolution and refactoring while reducing the risk of regressions
+
+## Evidence and Measures
+
+Integrating UI Tests in to your CI/CD is necessary but more challenging than unit tests.  The increased challenge is that UI tests either need to run headlessly with something like [Puppeteer](https://github.com/puppeteer/puppeteer) or there needs to be more extensive orchestration with Azure DevOps or Github that would handle the full testing integration for you like [BrowserStack](https://www.browserstack.com/automate/azure)
```

Comment:
[Playwright](https://github.com/microsoft/playwright) is worth a mention as well.

Reply:
nice! That's new to me, adding it.